### ISSUE TITLE
CA serial number conflicts possible

### ISSUE THREAD

**Issue by liggitt** [openshift/origin-metrics#47]
This line is problematic:
```
echo 03 > $dir/ca.serial.txt # otherwise openssl chokes on the file
```

1. It generates certificates starting with serial number 3, which will conflict with other certificates signed early on by the master CA. If a browser (or possibly other clients) ever hit a serving certificate with a conflicting serial from the same CA, it will disallow the certificate.

2. If modifies a local copy of the serial.txt, which means the serial numbers it signs certs for could (later) be reused by the master CA.

Is there a way to avoid injecting the CA and having the deployer sign certs locally, which the parent serial.txt file will have no record of?

**Comment by mwringe** [comment]
Good catch, but its actually not possible to send in the CA certificate and key to have it generate the certificates for you. At one point it did since I was using code originally from the aggregated logging, but that feature was removed because of the serial number conflict option.

It looks like some vestigial code got left behind (https://github.com/openshift/origin-metrics/blob/master/deployer/run.sh#L57 and https://github.com/openshift/origin-metrics/blob/master/deployer/run.sh#L138). But Line 58 and 59 are invalid and you cannot actually pass the CA and key in a secret and use those even if you want to (it will fail).

I will clean up those section of the code which didn't get removed before.

**Comment by liggitt** [comment]
ah, great. I'll leave it open to track the cleanup, thanks.

**Comment by mwringe** [comment]
yep, please leave it open until I get the PR committed. Otherwise it might get lost.

Thanks for pointing this out.

**Comment by mwringe** [comment]
The code in question is already being modified for another issue to add in support for a standalone Heapster install (for HPA). The fixes for this have already been added to the PR: https://github.com/openshift/origin-metrics/pull/46

Will close this issue once pull 46 is accomplished.

**Comment by mwringe** [comment]
Fixed with PR #46
